### DISCUSSION TITLE
[Feature Request?]: Support for multi tenant architectures?

### DISCUSSION THREAD

**Discussion by askides** [onejs/one#615] [Ideas]
Sometimes for architectural reasons you need to structure your app with a domain and many subdomains:

1. https://example.com
2. https://app.example.com
3. https://user.example.com

This is currently something feasible only in Next.js as even Remix does not support it natively.

Next.js [leverages on his middleware](https://github.com/vercel/platforms/blob/eeabcf1c8e29f5db86c15d71e98b8eded43bd341/middleware.ts) to handle such a feature.

The question is, are you guys in plan to build a general middleware to handle this kind of solutions? This would be really a game changer for me and finally a way to move out from Next.js which is definitely became too complex to handle for simple stuff.

I would love to contribute in case.

Extra docs: https://vercel.com/guides/nextjs-multi-tenant-application

**Comment by natew** [comment]
Wondering what would we need to add to middlewares to support this now that it's out:

https://onestack.dev/docs/routing-middlewares
